modal/cli/run.py

```diff
@@ -223,7 +224,10 @@ def _get_click_command_for_cls(app: App, method_ref: MethodReference):
     cls = method_ref.cls
     method_name = method_ref.method_name
 
-    cls_signature = _get_signature(cls._get_user_cls())
+    user_cls = cls._get_user_cls()
+    type_hints = safe_get_type_hints(user_cls)
+    sig: inspect.Signature = _get_class_constructor_signature(user_cls)
```

Comment:
We already had this utility for getting the signature of a modal class (regardless if it has an explicit or implicit constructor), so with some refactoring we could use it here